terraform/ratingservice/configure_rating_db.sh

```diff
@@ -32,7 +32,13 @@ cloud_sql_proxy -instances=$DB_HOST=tcp:5432 -verbose=false &>/dev/null &
 cloud_proxy_pid=$!
 
 # wait until proxy establishes connection
-sleep 5
+connected=false
+while [ $connected = false ]; do
+    echo "Waiting for establishing connection to db..."
+    PGPASSWORD=$DB_PASSWORD psql "host=127.0.0.1 sslmode=disable dbname=$DB_NAME user=$DB_USERNAME" -q -c '\conninfo' &>/dev/null
+    [ $? -eq 0 ] && connected=true
```

Comment:
I am not sure that I understand this remark. Testing exit code is a way to interpret whether psql succeed to establish connection or not. Other way would be to download some custom application designated to probe the connection without exit code.
There are other cases when the non zero exit code is the only way to capture data. For example, to check whether App Engine application is created for a project.

Reply:
Great, if `$?` isn't needed, can you add `set -o errexit` to the top of your script so it will fail fast if anything goes wrong? (see the install script)